backend/endpoints/design_assistant.py

```
@@ -0,0 +1,720 @@
+        elif isinstance(e, ServerException):
+            # Check for 403 errors in ServerException
+            error_str = str(e).lower()
+            if "403" in error_str:
+                # Safely get ids if available
+                try:
+                    doc_id = ids.get("documentId", "unknown") if 'ids' in locals() else "unknown"
+                    workspace_id = ids.get("wvmid", "unknown") if 'ids' in locals() else "unknown"
+                    element_id = ids.get("elementId", "unknown") if 'ids' in locals() else "unknown"
+                    wvm = ids.get("wvm", "unknown") if 'ids' in locals() else "unknown"
+                    element_path = f"{doc_id}/{wvm}/{workspace_id}/e/{element_id}" if 'ids' in locals() else "unknown"
+                except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)